api/v1/services/AuthService.js

```diff
@@ -59,3 +70,71 @@ module.exports.verify = (token) => _Promise
   const message = error.message;
   throw new errors.UnprocessableRequestError(message);
 });
+
+module.exports.getGitSessionCodePath = () => githubAuthRedirect + GITHUB_CLIENT_ID;
+
+module.exports.requestGitAccessToken = (code) => {
+  let token;
+  let githubId;
+
+  return request({
+    uri: githubTokenPath,
+    qs: {
+      client_id: GITHUB_CLIENT_ID,
+      client_secret: GITHUB_CLIENT_SECRET,
+      code: code
+    },
+    headers: {
+      'Accept': 'application/json'
+    }
+  })
+  .then((body) => {
+    token = JSON.parse(body).access_token;
+    return module.exports.getGitHubAccountDetails(token);
+  })
+  .then((handle) => {
+    githubId = handle;
+    return User.findByGitHubHandle(handle);
+  })
+  .then((user) => {
+    if(_.isNull(user)) {
+      return module.exports.getGitHubAccountEmail(token)
+      .then((email) => UserService.createGitHubUser(email, githubId))
+      .then(() => token);
+    }
+
+    return token;
+  });
+};
+
+module.exports.getGitHubAccountDetails = (authToken) => request({
+  uri: githubUserPath,
+  qs: {
+    access_token: authToken
+  },
+  headers: {
+    'User-Agent': 'HackIllinois-API'
```

Comment:
I just read why they make you do this, and it's interesting. How about this goes into a config entry somewhere and is renamed to `HackIllinois` since that's our GitHub organization's name?